llama_index/embeddings/mistralai.py

```diff
@@ -0,0 +1,82 @@
+"""MistralAI embeddings file."""
+
+from typing import Any, List, Optional
+
+from llama_index.bridge.pydantic import Field
+from llama_index.callbacks.base import CallbackManager
+from llama_index.embeddings.base import DEFAULT_EMBED_BATCH_SIZE, BaseEmbedding
+
+
+class MistralAIEmbedding(BaseEmbedding):
+    """Class for MistralAI embeddings.
+
+    Args:
+        model_name (str): Model for embedding.
+            Defaults to "mistral-embed".
+
+        api_key (Optional[str]): API key to access the model. Defaults to None.
+    """
+
+    # Instance variables initialized via Pydantic's mechanism
+    mistralai_client: Any = Field(description="MistralAI client")
+
+    def __init__(
+        self,
+        model_name: str = "mistral-embed",
+        api_key: Optional[str] = None,
+        embed_batch_size: int = DEFAULT_EMBED_BATCH_SIZE,
+        callback_manager: Optional[CallbackManager] = None,
+        **kwargs: Any,
+    ):
+        try:
+            from mistralai.client import MistralClient
+        except ImportError:
+            raise ImportError(
+                "mistralai package not found, install with" "'pip install mistralai'"
+            )
+        super().__init__(
+            mistralai_client=MistralClient(api_key=api_key),
+            model_name=model_name,
+            embed_batch_size=embed_batch_size,
+            callback_manager=callback_manager,
+            **kwargs,
+        )
+
+    @classmethod
+    def class_name(cls) -> str:
+        return "MistralAIEmbedding"
+
+    def _get_query_embedding(self, query: str) -> List[float]:
+        """Get query embedding."""
+        return (
+            self.mistralai_client.embeddings(model=self.model_name, input=[query])
+            .data[0]
+            .embedding
+        )
+
+    async def _aget_query_embedding(self, query: str) -> List[float]:
+        """The asynchronous version of _get_query_embedding."""
+        return self._get_query_embedding(query)
```

Comment:
I guess the client doesn't have async yet?

Reply:
Yes.